src/tests/field.rs

```diff
@@ -1,309 +1,808 @@
-use crate::serde::SerdeObject;
-use crate::{ff::Field, ff_ext::Legendre};
-use ark_std::{end_timer, start_timer};
-use rand::{RngCore, SeedableRng};
-use rand_xorshift::XorShiftRng;
-
-#[cfg(feature = "derive_serde")]
-use serde::{Deserialize, Serialize};
-
-pub fn random_field_tests<F: Field>(type_name: String) {
-    let mut rng = XorShiftRng::from_seed([
-        0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06, 0xbc,
-        0xe5,
-    ]);
-
-    random_multiplication_tests::<F, _>(&mut rng, type_name.clone());
-    random_addition_tests::<F, _>(&mut rng, type_name.clone());
-    random_subtraction_tests::<F, _>(&mut rng, type_name.clone());
-    random_negation_tests::<F, _>(&mut rng, type_name.clone());
-    random_doubling_tests::<F, _>(&mut rng, type_name.clone());
-    random_squaring_tests::<F, _>(&mut rng, type_name.clone());
-    random_inversion_tests::<F, _>(&mut rng, type_name.clone());
-    random_expansion_tests::<F, _>(&mut rng, type_name);
-
-    assert_eq!(F::ZERO.is_zero().unwrap_u8(), 1);
-    {
-        let mut z = F::ZERO;
-        z = z.neg();
-        assert_eq!(z.is_zero().unwrap_u8(), 1);
-    }
-
-    assert!(bool::from(F::ZERO.invert().is_none()));
-
-    // Multiplication by zero
-    {
-        let mut a = F::random(&mut rng);
-        a.mul_assign(&F::ZERO);
-        assert_eq!(a.is_zero().unwrap_u8(), 1);
-    }
-
-    // Addition by zero
-    {
-        let mut a = F::random(&mut rng);
-        let copy = a;
-        a.add_assign(&F::ZERO);
-        assert_eq!(a, copy);
-    }
-}
+#[macro_export]
+macro_rules! field_testing_suite {
+    ($field: ident, "field_arithmetic") => {
+        macro_rules! random_multiplication_tests {
+            ($f: ident, $rng: expr) => {
+                let _message = format!("multiplication {}", stringify!($f));
+                let start = start_timer!(|| _message);
```

Comment:
Remove the unnecessary `timer`s in https://github.com/privacy-scaling-explorations/halo2curves/pull/131/commits/b549998388ed609dc1682dd7a9223ad79c991868

Regarding the `ark-std` dep, it is used in several places.
For example, it is for feature [`print-trace`](https://github.com/privacy-scaling-explorations/halo2curves/blob/main/Cargo.toml#L51),  [`msm testing`](https://github.com/privacy-scaling-explorations/halo2curves/blob/main/src/msm.rs#L188), and [`bn256::Fr benchmarking`](https://github.com/privacy-scaling-explorations/halo2curves/blob/main/src/bn256/fr.rs#L457-L466) 
I think we should keep `ark-std` because of these usages.